cfme/tests/infrastructure/test_advanced_search_vms.py

```diff
@@ -333,3 +341,70 @@ def test_cannot_delete_vm_filter_more_than_once(vm_view):
     vm_view.flash.assert_no_error()
     # Try it second time
     assert not vm_view.entities.search.delete_filter(), 'Delete twice accepted!'
+
+
+@pytest.mark.tier(1)
+@pytest.mark.meta(automates=[1506672, 1660460, 1718895])
+def test_create_filter_with_multiple_conditions(appliance, provider, request, running_vms):
+    """
+    Polarion:
+        assignee: anikifor
+        casecomponent: WebUI
+        caseimportance: high
+        initialEstimate: 1/5h
+        setup:
+            1. Navigate to Compute > Infrastructure > Providers.
+            2. Click on Advanced Search Filter.
+        testSteps:
+            1. Create an expression with multiple types of condition. Eg: arg_1 AND arg_2 OR arg_3
+        expectedResults:
+            1. Expression must be created successfully.
+
+    Bugzilla:
+        1506672
+        1660460
+        1718895
+    """
+    filter_name = fauxfactory.gen_alphanumeric()
+
+    if len(running_vms) >= 2:
+        vm1_name, vm2_name = running_vms[0], running_vms[1]
+        vm1 = appliance.collections.infra_vms.instantiate(vm1_name, provider)
+        vm2 = appliance.collections.infra_vms.instantiate(vm2_name, provider)
+    else:
+        pytest.skip("Less than 2 VMs are running, that's not enough for multiple condition test")
+
+    # add the same tags as in the customer case
+    tag1 = (appliance.collections.categories.instantiate(display_name="Environment")
+            .collections.tags.instantiate(display_name="Development"))
+    tag2 = (appliance.collections.categories.instantiate(display_name="Environment")
+            .collections.tags.instantiate(display_name="Test"))
+    vm1.add_tag(tag1)
+    vm2.add_tag(tag2)
+
+    vm_view = navigate_to(appliance.collections.infra_vms, 'VMsOnly')
+    # Find the number of VMs before filtering
+    vms_before = len(vm_view.entities.get_all())
```

Comment:
I would use `vm_view.paginator.items_amount` attribute here to get the number of items on all the pages. This call to `get_all` is costly because its creating python object instances for each item on the pages.